ddpui/api/dbt_api.py

```diff
@@ -151,6 +157,126 @@ def put_dbt_github(request, payload: OrgDbtGitHub):
+
+    # Determine which PAT to use for verification
+    if is_token_masked:
+        # Token is masked, retrieve the actual PAT from secrets manager
+        if not orgdbt.gitrepo_access_token_secret:
+            raise HttpError(
+                400, "Cannot use masked token - no existing PAT found. Please provide a valid PAT."
+            )
+        actual_pat = secretsmanager.retrieve_github_pat(orgdbt.gitrepo_access_token_secret)
+        if not actual_pat:
+            raise HttpError(400, "Failed to retrieve existing PAT from secrets manager")
+    else:
+        actual_pat = payload.gitrepoAccessToken
+
+    # Validate git is initialized locally
+    try:
+        git_manager = GitManager(
+            repo_local_path=str(dbt_repo_dir), pat=actual_pat, validate_git=True
+        )
+    except GitManagerError as e:
+        raise HttpError(400, f"Git is not initialized in the DBT repo: {e.message}") from e
+
+    # Verify remote URL is accessible with the PAT
+    try:
+        git_manager.verify_remote_url(payload.gitrepoUrl)
+    except GitManagerError as e:
+        raise HttpError(400, f"{e.message}: {e.error}") from e
+
+    # Set or update the remote origin
+    try:
+        git_manager.set_remote(payload.gitrepoUrl)
+    except GitManagerError as e:
+        raise HttpError(500, f"Failed to set remote: {e.message}") from e
+
+    # Handle PAT token storage (only if not masked)
+    if not is_token_masked:
+        # Create oauth URL for prefect secret block
+        gitrepo_url_with_token = GitManager.generate_oauth_url_static(
+            payload.gitrepoUrl, payload.gitrepoAccessToken
+        )
+
+        # Create or update the prefect secret block
+        secret_block_edit_params = PrefectSecretBlockEdit(
+            block_name=f"{org.slug}-git-pull-url",
+            secret=gitrepo_url_with_token,
+        )
+
+        response = prefect_service.upsert_secret_block(secret_block_edit_params)
+        if not OrgPrefectBlockv1.objects.filter(
+            org=org, block_type=SECRET, block_name=secret_block_edit_params.block_name
+        ).exists():
+            OrgPrefectBlockv1.objects.create(
+                org=org,
+                block_type=SECRET,
+                block_name=secret_block_edit_params.block_name,
+                block_id=response["block_id"],
+            )
+
+        # Update or create PAT in secrets manager
+        if orgdbt.gitrepo_access_token_secret:
+            secretsmanager.update_github_pat(
+                orgdbt.gitrepo_access_token_secret, payload.gitrepoAccessToken
+            )
+        else:
+            pat_secret_key = secretsmanager.save_github_pat(payload.gitrepoAccessToken)
+            orgdbt.gitrepo_access_token_secret = pat_secret_key
+
+    # Update OrgDbt with the new gitrepo_url
+    orgdbt.gitrepo_url = payload.gitrepoUrl
+    orgdbt.save()
+
+    logger.info(f"Connected git remote for org {org.slug}: {payload.gitrepoUrl}")
+
+    return {
```

Comment:
since we are in the midst of standardizing our codebase & refactoring.

do you want to come up with some way of returninig a standardized api response that can eventually be used across the entire codebase ? will leave up to you if you want to do in this PR or later.